tests/common-lib/cliClient.js

```diff
@@ -12,4 +18,92 @@ const execCliCmdString = async (commands) => {
   })
 }
 
+/**
+ * Query the Kubernetes API using the oc CLI to get the expected state.
+ * @param string kind
+ * @param string apigroup
+ * @param {[string]} namespace
+ * @param {*} cluster
+ * @returns {[]} Resources
+ */
+function getResourcesFromOC({
+  user,
+  kind,
+  apigroup,
+  namespace = '--all-namespaces',
+  cluster = { type: 'hub', name: 'local-cluster' },
+}) {
+  if (!kind) {
+    console.error('Error in test code, kind is required when calling getResourcesFromOC(). Received:', kind)
+  }
+  var property = kind
+
+  // Check to see if the test needs to include the apigroup name within the query.
+  // For kind resources with v1 versions, no apigroup is needed.
+  if (apigroup && apigroup.useAPIGroup && apigroup.name != 'v1') property += `.${apigroup.name}`
+
+  var cmd = `oc get ${property.toLowerCase()} ${
+    namespace === '--all-namespaces' ? namespace : `-n ${namespace}`
+  } --ignore-not-found=true`
+
+  // Add kubeconfig filter if the option is set within the cluster object.
+  if (cluster.kubeconfig) {
+    cmd += ` --kubeconfig ${cluster.kubeconfig}`
+  }
+
+  // Impersonate the user.
+  if (user && (user.fullName || user.name)) {
+    cmd += ` --as=${user.fullName || user.name}`
+  }
+
+  try {
+    return formatListOutput(
+      cluster,
+      kind,
+      execSync(cmd, { stdio: [] })
+        .toString()
+        .split('\n')
+        .filter((res) => res)
+    )
+  } catch (err) {
+    if (err.message.indexOf("the server doesn't have a resource type") > 0) {
+      // This is expected when a CRD gets removed by another test.
+      return []
+    } else if (err.message.indexOf('Error from server (Forbidden)') > 0) {
+      // This is expected when user ddoesn't have access to a resource.
+      return []
+    }
+    console.log(`Unexpected error getting resources from CLI. ${err.message}`)
+    throw err
+  }
+}
+
+/**
+ * Format resources for search queries.
+ * @param {Object} cluster The cluster of the resource objects.
+ * @param {string} kind The kind of the resource objects.
+ * @param {[string]} resources An array of strings that will be formated as an object containing name and namespace.
+ * @returns `formatedResources` Formatted array of resource object.
+ */
+function formatListOutput(cluster, kind, resources) {
+  let headers, nameIndex, namespaceIndex
+  if (resources && resources.lenght > 1) {
+    headers = resources.shift().split(' ').trim().toLowerCase()
+    nameIndex = headers.indexOf('name')
+    namespaceIndex = headers.indexOf('namespace')
+  }
```

Comment:
Made this generic, it now uses the headers to find the position of the `name` and `namespace` fields in the output.